packages/react/src/components/Header/HeaderActionGroup.jsx

```diff
@@ -16,19 +24,146 @@ const propTypes = {
+      const nameRect = overFlowContainerRef.current?.previousSibling?.getBoundingClientRect();
+      /* istanbul ignore else */
+      if (actionRect && nameRect) {
+        const windowWidth = window.innerWidth || document.documentElement.clientWidth;
+        // check that it's also greater than zero to prevent collapsing in jest where all the values are 0.
+        const tooBig = nameRect.right > 0 && nameRect.right >= actionRect.left;
+        const previousBreakpoint = breakpoint.current;
+
+        if (tooBig && actionItems.length > 0 && overflowItems.length === 0) {
+          breakpoint.current = windowWidth;
+          setOverflowItems(actionItems);
+        } else if (windowWidth > previousBreakpoint) {
+          setOverflowItems([]);
+        }
+      }
+    }
+  }, [actionItems, overflowItems.length]);
+
+  useLayoutEffect(() => {
+    checkForOverflow();
+    window.addEventListener('resize', checkForOverflow);
+
+    return () => window.removeEventListener('resize', checkForOverflow);
+  }, [checkForOverflow]);
+
   return (
-    <>
+    // added ever div here, because HeaderGlobalBar doesn't support refs
+    <div ref={overFlowContainerRef} className={`${prefix}--header__global`}>
       <HeaderGlobalBar>
-        {actionItems.map((item, i) => (
-          <HeaderAction
-            item={item}
-            index={i}
-            key={`header-action-item-${item.label}-${i}`}
-            testID={`header-action-item-${item.label}`}
-          />
-        ))}
+        {
+          // if we have overflow items and are not showing a header action submenu
+          // then render the overflow menu
+          overflowItems.length > 0 && !showMenu ? (
+            <div
+              className={`${iotPrefix}--header__overflow-menu-container`}
+              data-floating-menu-container
+            >
+              <OverflowMenu
+                useAutoPositioning
+                onOpen={() => setOverflowOpen(true)}
+                onClose={() => setOverflowOpen(false)}
+                renderIcon={() =>
+                  // show a close icon when open per design specs
+                  overflowOpen ? <Close16 fill="white" /> : <OverflowMenuVertical16 fill="white" />
+                }
+              >
+                {overflowItems.map((child, i) => {
+                  if (
+                    child.hasHeaderPanel ||
+                    (Array.isArray(child.childContent) && child.childContent.length > 0)
+                  ) {
+                    ReactDOM.createPortal(
+                      [
+                        <HeaderAction
+                          item={child}
+                          index={i}
+                          key={`header-action-item-${child.label}-${i}`}
+                          testID={`header-action-item-${child.label}`}
+                          // used to render only the label in the overflow menu instead of the icon
+                          inOverflow
```

Comment:
why not use the name "renderLabel" for this prop instead?

Reply:
Absolutely. #namingishard.